### PR TITLE
Player v7 | Live 2.0.7| - Entry view - While the stream is in preview an error will be shown in the player

### PR DESCRIPTION
**Pull request by semarche-kaltura**
1) removed decorator
2) render offline states with player-ui API
3) render live tag with player-ui API
4) handle play,seek,pause events in listeners
5) removed REPLY button
6) changed prettier settings for brackets

### CODE REVIEW

File: src/components/live-tag/live-tag.tsx [hunk 1]
```diff
@@ -7,49 +7,111 @@ export enum LiveTagStates {
...
+    this.setState({
+      liveTagState: state,
+    });
+  };
+
+  private _updateIsOnLiveEdge = () => {
+    this.setState({
+      isOnLiveEdge: this.context.player.paused
+        ? false
+        : this.context.player.isOnLiveEdge(),
```
Comment (yairans) [inline src/components/live-tag/live-tag.tsx:78]: `player.isOnLiveEdge()` already should return `false` for `paused`. is not?

File: src/components/live-tag/live-tag.tsx [hunk 2]
```diff
@@ -7,49 +7,111 @@ export enum LiveTagStates {
...
-    if (isOnLiveEdge && state === LiveTagStates.Live) {
+    const { isOnLiveEdge, liveTagState } = this.state;
+    if (isOnLiveEdge && liveTagState === LiveTagStates.Live) {
       return styles.live;
     }
-    if (isOnLiveEdge && state === LiveTagStates.Preview) {
+    if (isOnLiveEdge && liveTagState === LiveTagStates.Preview) {
       return styles.preview;
     }
     return styles.offline;
```
Comment (yairans) [inline src/components/live-tag/live-tag.tsx:99]: I would change the style names (e.g. liveEdge, previewEdge, nonEdge)

File: src/components/offline-slate/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from "./offline-slate";
```
Comment (yairans) [inline src/components/offline-slate/index.ts:1]: seems prettier doesn't catch the double quotes

File: src/components/no-longer-live/no-longer-live.tsx [hunk 4]
```diff
@@ -1,37 +1,19 @@
...
-    render(props: NoLongerLiveProps) {
-        return (
-            <div className={styles.noLongerLiveWrapper}>
-                <div className={`${styles.iconWrapper} kaltura-live__icon-wrapper`}>
-                    <div
-                        className={`${styles.noLongerLiveIcon} kaltura-live__no-longer-live-icon`}
-                    />
-                    {props.showReplay && (
-                        <a
-                            className={styles.replayIconWrapper}
```
Comment (yairans) [inline src/components/no-longer-live/no-longer-live.tsx:19]: remove from css file

File: src/components/offline-slate/offline-slate.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  render() {
+    if (this.state.type === OfflineTypes.HttpError) {
+      return <Offline httpError={true} />;
+    }
+    if (this.state.type === OfflineTypes.NoLongerLive) {
+      return <NoLongerLive />;
+    }
+    return <Offline />;
+  }
```
Comment (yairans) [inline src/components/offline-slate/offline-slate.tsx:44]:
```suggestion
  render() {
    if (this.state.type === OfflineTypes.NoLongerLive) {
      return <NoLongerLive />;
    }
    return <Offline httpError={this.state.type === OfflineTypes.HttpError} />;
  }```

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/components/live-tag/live-tag.tsx (hunk 1 above)
Comment (semarche-kaltura) [reply src/components/live-tag/live-tag.tsx:78]: No, the player.isOnLiveEdge() returns true for paused. I achieved result only with setTimeout
`
setTimeout(() => {
      this.setState({
        isOnLiveEdge: this.context.player.isOnLiveEdge(),
      });
    });
`

Comment (yairans) [reply src/components/live-tag/live-tag.tsx:78]: totally agree it's better than timeout, but seems it's bug in the core impl. 
can you investigate why [this line](https://github.com/kaltura/playkit-js/blob/v0.67.0/src/player.js#L2052) doesn't work?

Comment (semarche-kaltura) [reply src/components/live-tag/live-tag.tsx:78]: listener for `PAUSE` event in kaltura-live plugin triggers earlier than listener in player.js
https://github.com/kaltura/playkit-js/blob/master/src/player.js#L1731
that's why `_isOnLiveEdge ` property is true

**Review by yairans — APPROVED** [review]

File: src/components/live-tag/live-tag.tsx [hunk 6]
```diff
@@ -13,30 +13,30 @@ interface LiveTagProps {
...
 }
 
 export class LiveTag extends Component<LiveTagProps, LiveTagState> {
-  constructor(props: LiveTagProps, context: Context) {
+  constructor(props: LiveTagProps, { player }: Context) {
     super();
     this.state = {
       liveTagState: props.liveTagState,
-      isOnLiveEdge: context.player.isOnLiveEdge(),
+      behindLiveEdge: player.isDvr && !player.isOnLiveEdge(),
```
Comment (yairans) [inline src/components/live-tag/live-tag.tsx:28]: just call to `this._updateIsOnLiveEdge()` for reuse

Comment (yairans) [reply src/components/live-tag/live-tag.tsx:28]: actually i would move all staff here to componentDidMount

**Review by yairans — CHANGES_REQUESTED** [review]

**Comment by yairans** [comment]
decorator is required. moved to https://github.com/kaltura/playkit-js-kaltura-live/pull/248

**Pull request closed without merging**